programs/server/Server.cpp

```diff
@@ -1117,6 +1118,8 @@ try
     /// NOTE: global context should be destroyed *before* GlobalThreadPool::shutdown()
     /// Otherwise GlobalThreadPool::shutdown() will hang, since Context holds some threads.
     SCOPE_EXIT({
+        CurrentMetrics::set(CurrentMetrics::IsServerShuttingDown, 1);
```

Comment:
I think it would be more correct to move it to the first scope_exit which will be executed on shutdown, and it would be this one https://github.com/ClickHouse/ClickHouse/blob/a170aac450f14c3f06ab061a02eb4783f6088f03/programs/server/Server.cpp#L2529

Reply:
done